wordfreq_builder/wordfreq_builder/tokenizers.py

```diff
@@ -26,48 +27,63 @@
 MARKDOWN_URL_RESIDUE_RE = regex.compile(r'\]\(\)')
 
 
-def cld2_surface_tokenizer(text):
-    """
-    Uses CLD2 to detect the language and wordfreq tokenizer to create tokens.
-    """
-    text = unescape_html(text)
-    text = TWITTER_HANDLE_RE.sub('', text)
-    text = TCO_RE.sub('', text)
-
-    lang = cld2_detect_language(text)
-
-    # Don't allow tokenization in Chinese when language-detecting, because
-    # the Chinese tokenizer may not be built yet
-    if lang == 'zh':
-        lang = 'en'
-
-    tokens = tokenize(text, lang)
-    return lang, tokens
+# Low-frequency languages tend to be detected incorrectly by cld2. The
+# following list of languages are languages that appear in our data with any
+# reasonable frequency, and seem to usually be detected *correctly*. These are
+# the languages we'll keep in the Reddit and Twitter results.
+#
+# This list is larger than the list that wordfreq ultimately generates, so we
+# can look here as a source of future data.
 
-
-# Low-frequency languages tend to be detected incorrectly. Keep a limited
-# list of languages we're allowed to use here.
 KEEP_THESE_LANGUAGES = {
-    'ar', 'de', 'el', 'en', 'es', 'fr', 'hr', 'id', 'it', 'ja', 'ko', 'ms',
-    'nl', 'pl', 'pt', 'ro', 'ru', 'sv'
+    'af', 'ar', 'bs', 'ca', 'cs', 'da', 'de', 'el', 'en', 'es', 'et', 'fi',
+    'fr', 'gl', 'he', 'hi', 'hr', 'hu', 'id', 'is', 'it', 'ja', 'ko', 'lv',
+    'ms', 'nl', 'nn', 'no', 'pl', 'pt', 'ro', 'ru', 'sr', 'sv', 'sw', 'tl',
+    'tr', 'uk', 'vi'
 }
 
-
-def cld2_reddit_tokenizer(text):
+# Semi-frequent languages that are excluded by the above:
+#
+#   - Chinese, not because it's detected incorrectly, but because we can't
+#     handle it until we already have word frequencies
+#   - Thai (seems to be detected whenever someone uses Thai characters in
+#     an emoticon)
+#   - Welsh (which is detected for "ohmygodohmygodohmygod")
+#   - Turkmen (detected for ASCII art)
+#   - Irish Gaelic (detected for Cthulhu-related text)
+#   - Kannada (looks of disapproval)
+#   - Lao, Tamil, Xhosa, Slovak (various emoticons and Internet memes)
+#   - Breton (the word "memes" itself)
+
+
+def cld2_surface_tokenizer(text, mode='twitter'):
     """
-    A language-detecting tokenizer with special cases for handling text from
-    Reddit.
+    Uses CLD2 to detect the language and wordfreq tokenizer to create tokens.
+
+    The `mode` can be 'twitter' or 'reddit', which slightly changes the
+    pre-processing of the text.
     """
-    text = URL_RE.sub('', text)
-    text = MARKDOWN_URL_RESIDUE_RE.sub(']', text)
+    text = unescape_html(text)
+    if mode == 'twitter':
+        text = TWITTER_HANDLE_RE.sub('', text)
+        text = TCO_RE.sub('', text)
+    elif mode == 'reddit':
+        text = URL_RE.sub('', text)
+        text = MARKDOWN_URL_RESIDUE_RE.sub(']', text)
```

Comment:
Seems reasonable to me. If we manage to feed it something that isn't Twitter or Reddit, it shouldn't do Twitter or Reddit-specific text fixes.